traverson4j-hc5/src/main/java/uk/co/autotrader/traverson/http/ApacheHttpTraversonClientAdapter.java

```diff
@@ -37,32 +37,42 @@ public ApacheHttpTraversonClientAdapter(CloseableHttpClient client) {
 
     @Override
     public <T> Response<T> execute(Request request, Class<T> returnType) {
-        HttpUriRequest httpUriRequest = apacheHttpUriConverter.toRequest(request);
+        ClassicHttpRequest httpRequest = apacheHttpUriConverter.toRequest(request);
```

Comment:
Unfortunately this change is now breaking checkstyle for being too complex. I think we need to redesign this space and improve the testability of the code. If you'd want to pen and paper with me on how we should take this forward, I'd be happy to work with you :)

Reply:
Yeah that sounds good. I did move the for loop sectiion into it's own method but I reverted it as I didnt think it was the best solution

Reply:
done

Reply:
or at least had a stab at it :)